### PR TITLE
Add feature: systemwide memory limit in snapshotter.

### PR DESCRIPTION
**Pull request by NorCalRobotics**

### CODE REVIEW

**Comment by JWhitleyWork** [comment]
@JackAtGaia / @LaurentiuCristofor / @stevegaia Would you guys mind taking a look at this feature addition? The dev says it's "a little hacky" so he wanted a few more eyeballs on it.

**Comment by LaurentiuCristofor** [comment]
> @JackAtGaia / @LaurentiuCristofor / @stevegaia Would you guys mind taking a look at this feature addition? The dev says it's "a little hacky" so he wanted a few more eyeballs on it.

Keep in mind that I'm not very familiar with ROS and this project. Can you describe this feature at a high level? Also, if there are particular areas of code that you want reviewed, can you point them out with comments (PR comments, not code comments)?

File: rosbag2_snapshot/src/snapshotter.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@
 #include <rclcpp/rclcpp.hpp>
 #include <rosbag2_snapshot/snapshotter.hpp>
 
-#include <filesystem>
+#include <experimental/filesystem>
```
Comment (NorCalRobotics) [inline rosbag2_snapshot/src/snapshotter.cpp:37]: @JWhitleyWork are you aware of best practices for how to handle this include path / namespace issue with std::filesystem vs std::experimental::filesystem on differing OS'es? I would like this to include whichever header is available at compile time, and then use the correct namespace, to make this code more cross-platform.

File: rosbag2_snapshot/src/snapshotter.cpp [hunk 2]
```diff
@@ -791,7 +961,7 @@ void SnapshotterClient::setSnapshotterClientOptions(const SnapshotterClientOptio
     if (req->filename.empty()) {
       req->filename = "./";
     }
-    std::filesystem::path p(std::filesystem::absolute(req->filename));
+    std::experimental::filesystem::path p(std::experimental::filesystem::absolute(req->filename));
```
Comment (NorCalRobotics) [inline rosbag2_snapshot/src/snapshotter.cpp]: This is where the namespace corresponding to #include \<filesystem\> is used in the code.

**Comment by vDonGlover** [comment]
Note to reviewers: Please do not merge this PR until we have finalized our agreements for external contributions.

**Pull request closed without merging**